packages/in_app_purchase/in_app_purchase_android/lib/src/billing_client_wrappers/billing_client_manager.dart

```diff
@@ -0,0 +1,155 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/widgets.dart';
+
+import 'billing_client_wrapper.dart';
+import 'purchase_wrapper.dart';
+
+/// Abstraction of result of [BillingClient] operation that includes
+/// a [BillingResponse].
+abstract class HasBillingResponse {
+  /// The status of the operation.
+  abstract final BillingResponse responseCode;
+}
+
+/// Utility class that manages a [BillingClient] connection.
+///
+/// Connection is initialized on creation of [BillingClientManager].
+/// If [BillingClient] sends `onBillingServiceDisconnected` event or any
+/// operation returns [BillingResponse.serviceDisconnected], connection is
+/// re-initialized.
+///
+/// [BillingClient] instance is not exposed directly. It can be accessed via
+/// [runWithClient] and [runWithClientNonRetryable] methods that handle the
+/// connection management.
+///
+/// Consider calling [dispose] after the [BillingClient] is no longer needed.
+class BillingClientManager {
+  /// Creates the [BillingClientManager].
+  ///
+  /// Immediately initializes connection to the underlying [BillingClient].
+  BillingClientManager() {
+    _connect();
+  }
+
+  /// Stream of `onPurchasesUpdated` events from the [BillingClient].
+  ///
+  /// This is a broadcast stream, so it can be listened to multiple times.
+  /// A "done" event will be sent after [dispose] is called.
+  late final Stream<PurchasesResultWrapper> purchasesUpdatedStream =
+      _purchasesUpdatedController.stream;
+
+  /// [BillingClient] instance managed by this [BillingClientManager].
+  ///
+  /// In order to access the [BillingClient], consider using [runWithClient]
+  /// and [runWithClientNonRetryable]
+  /// methods.
```

Comment:
Since this is only visible for testing, it should recommend only using these methods.

nit:
```suggestion
  /// In order to access the [BillingClient], use [runWithClient]
  /// or [runWithClientNonRetryable] methods.
```